src/django_components/util/loader.py

```diff
@@ -211,13 +211,11 @@ def _filepath_to_python_module(
     - Then the path relative to project root is `app/components/mycomp.py`
     - Which we then turn into python import path `app.components.mycomp`
     """
-    rel_path = os.path.relpath(file_path, start=root_fs_path)
-    rel_path_without_suffix = str(Path(rel_path).with_suffix(""))
+    path_cls = PureWindowsPath if os.name == "nt" else PurePosixPath
```

Comment:
Refactored to rely on `os.name` instead of `os.path.sep` to decide which separator we expect.